deploifai/api/__init__.py

```diff
@@ -410,3 +409,213 @@ def create_project(self, project_name: str, cloud_profile: CloudProfile, fragmen
+            )
+            api_data = r.json()
+
+            return api_data["data"]["falconMLConfigs"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+
+    def create_training_server(self, name: str, project_id: str, cloud_profile_id: str,
+                               falcon_plan: str, falcon_gpu: str,
+                               data_storage_id: str = None, falcon_id: str = None):
+        mutation = """
+        mutation($whereProject: ProjectWhereUniqueInput! $data: CreateTrainingInput!){
+            createTraining(whereProject: $whereProject data: $data){
+            name
+            status
+            }
+        }
+        """
+
+        if falcon_id:
+            if data_storage_id:
+                variables = {
+                    "whereProject": {"id": project_id},
+                    "data": {
+                        "name": name,
+                        "cloudProfileId": cloud_profile_id,
+                        "cloudProviderFalconConfig": {
+                            "plan": falcon_plan,
+                            "usesGpu": falcon_gpu,
+                        },
+                        "falconMLConfigId": falcon_id,
+                        "dataStorageIds": data_storage_id,
+                    },
+                }
+            else:
+                variables = {
+                    "whereProject": {"id": project_id},
+                    "data": {
+                        "name": name,
+                        "cloudProfileId": cloud_profile_id,
+                        "cloudProviderFalconConfig": {
+                            "plan": falcon_plan,
+                            "usesGpu": falcon_gpu,
+                        },
+                        "falconMLConfigId": falcon_id,
+                    },
+                }
+        else:
+            if data_storage_id:
+                variables = {
+                    "whereProject": {"id": project_id},
+                    "data": {
+                        "name": name,
+                        "cloudProfileId": cloud_profile_id,
+                        "cloudProviderFalconConfig": {
+                            "plan": falcon_plan,
+                            "usesGpu": falcon_gpu,
+                        },
+                        "dataStorageIds": data_storage_id,
+                    },
+                }
+            else:
+                variables = {
+                    "whereProject": {"id": project_id},
+                    "data": {
+                        "name": name,
+                        "cloudProfileId": cloud_profile_id,
+                        "cloudProviderFalconConfig": {
+                            "plan": falcon_plan,
+                            "usesGpu": falcon_gpu,
+                        },
+                    },
+                }
```

Comment:
Look into python dictionary updates to add / modify key-value pairs in a dictionary. This level of conditional is a bit too redundant, but it's just my opinion.

Reply:
Just passing none in the values also works
so we don't need the if, else conditions here